flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionJobVertex.java

```diff
@@ -82,6 +82,8 @@
 	
 	private InputSplitAssigner splitAssigner;
 	
+	private boolean receivedStopSignal = false;
```

Comment:
Don't we want to reset the stop signal in case of a restart of the `ExecutionJobVertex`?